rpc_jobs/hardening.yml

```diff
@@ -0,0 +1,133 @@
+          default: "{USER_VARS}"
+          description: "OSA/RPC USER_VARS to inject for this build"
+      - string:
+          name: STAGES
+          default: |
+            Allocate Resources, Connect Slave, Prepare Multi-Node AIO,
+             Prepare RPC Configs, Deploy RPC w/ Script, Cleanup,
+             Destroy Slave
+          description: |
+            Pipeline stages to run CSV. Note that this list does not influence execution order.
+            Options:
+              Allocate Resources
+              Connect Slave
+              Prepare Multi-Node AIO
+              Prepare RPC Configs
+              Deploy RPC w/ Script
+              Prepare MaaS
+              Setup MaaS
+              Verify MaaS
+              Install Tempest
+              Tempest Tests
+              Prepare Horizon Selenium
+              Horizon Tests
+              Prepare Kibana Selenium
+              Kibana Tests
+              Holland
+              Pause (use to hold instance for investigation before cleanup)
+              Cleanup
+              Destroy Slave
+
+    dsl: |
+      node("CentOS"){{
+        try {{
+          dir("rpc-gating"){{
+            git branch: env.RPC_GATING_BRANCH, url: env.RPC_GATING_REPO
+            pubCloudSlave = load 'pipeline_steps/pubcloud.groovy'
+            ssh_slave = load 'pipeline_steps/ssh_slave.groovy'
+            common = load 'pipeline_steps/common.groovy'
+            multi_node_aio_prepare = load 'pipeline_steps/multi_node_aio_prepare.groovy'
+            deploy = load 'pipeline_steps/deploy.groovy'
+            maas = load 'pipeline_steps/maas.groovy'
+          }} // dir
+
+          withCredentials([
+            string(
+              credentialsId: 'hardening_repo_url',
+              variable: 'HARDENING_REPO'
+            )
+          ]){{
+            dir("RPC-Unified-Gating") {{
+              git branch: "master", url: env.HARDENING_REPO, credentialsId: 'rpc-jenkins-svc-github-key'
+            }} // dir
+          }} // withCred
+          stash includes: 'RPC-Unified-Gating/', name: 'RPC-Unified-Gating'
+
+          instance_name = common.gen_instance_name()
+          deploy_node = null
+          pubCloudSlave.getPubCloudSlave(instance_name: instance_name)
+
+          node(instance_name){{
+            multi_node_aio_prepare.prepare()
+            instance_ip = sh(script: "ip route get 1 | awk '{{print \$NF;exit}}'", returnStdout: true)
+          }} // public cloud node
+
+          deploy_node = "${{instance_name}}-deploy-vm"
+          multi_node_aio_prepare.connect_deploy_node(deploy_node, instance_ip)
+
+          node(deploy_node){{
+            deploy.deploy_sh(
+              environment_vars: [
+                "DEPLOY_HAPROXY=yes",
+                "DEPLOY_ELK=yes",
+                "DEPLOY_TEMPEST=no",
+                "DEPLOY_AIO=no",
+                "DEPLOY_MAAS=no"
+                ]
+            ) // deploy_sh
+            unstash 'RPC-Unified-Gating'
```

Comment:
Why not run other tests while the node is up? While this would result in a longer run time, it could help to ensure the environment built as expected and results from the RPC-Unified-Gating tests were done on a good environment. Even a subset of Tempest such as just smoke tests could help to ensure valid results without the time of a full run.